documentation/public/user-guide/query-language.md

```diff
@@ -0,0 +1,109 @@
+# The Tree Query Language
+
+## Query Syntax
+
+A _query_ is a pattern that matches a
+certain set of nodes in a tree. The expression to match a given node
+consists of a pair of brackets (`[]`) containing two things: the node's kind, and
+optionally, a series of other patterns that match the node's children. For
+example, this pattern would match any `MultiplicativeExpression` node whose children
+are exactly both `Expression` nodes with an `Asterisk` node in between (no whitespace):
```

Comment:
nit suggestion: using `two` instead of `both`:

```suggestion
are exactly two `Expression` nodes, with an `Asterisk` node in between (no whitespace):
```

Reply:
thanks, done in https://github.com/NomicFoundation/slang/pull/928/commits/3c2f5016ff9336ce314ddc51942e86276f366c48